### PR TITLE
Star Rating

### PR DESCRIPTION
**Pull request by octavioamu**
##### Description
Star Rating implementations

##### Checklist

- [x] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [ ] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [x] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Refers/Fixes

Refs: #3879
Refs: #4001 
Refs: #3880 
Refs: #3964 
###### Contributor

- [ ] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Tested all changes **locally**
- [ ] Verified existing functionality
- [ ] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

File: app/dashboard/templates/shared/rate_bounties.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            <a class="nav-link nav-line" id="funded-tab" data-toggle="tab" href="#funded" role="tab" aria-controls="funded" aria-selected="false">{% trans "Funded" %} (7)</a>
+          </li>
+        </ul>
+        <div class="tab-content" id="myTabContent">
+          <div class="tab-pane fade show active" id="worksubmitted" role="tabpanel" aria-labelledby="home-tab">
+            <div class="scroll-carousel">
+              {% for i in "54321" %}
+                <div class="card-rating">
+                  <div class="card-rating__header">
+                    <img src="http://localhost:8000/dynamic/avatar/gitcoinco" alt="" width="56" height="56" class="card-rating__avatar rounded-circle">
```
Comment (rafalkowalski) [inline app/dashboard/templates/shared/rate_bounties.html]: I think that you shouldn't use URL with localhost.

File: app/dashboard/templates/shared/rating.html [hunk 2]
```diff
@@ -15,16 +15,24 @@
   along with this program. If not, see <http://www.gnu.org/licenses/>.
 {% endcomment %}
 {% load i18n %}
-<div class="w-100 mt-2" style="height: 70px;">
-<label class="form__label" for="notificationEmail">{% trans "Rate your experience with the " %}{{ counterparty }}</label>
-  <div class="rating">
+<div class="w-100 p-2">
+<!-- <label class="form__label" for="notificationEmail">{% trans "Rate your experience with the " %}{{ counterparty }}</label> -->
```
Comment (rafalkowalski) [inline app/dashboard/templates/shared/rating.html]: All unused commented code should be removed

Comment (octavioamu) [reply app/dashboard/templates/shared/rating.html]: yes thanks, is a WIP for now just some static front end.

Comment (rafalkowalski) [reply app/dashboard/templates/shared/rating.html]: OK :)

**Comment by thelostone-mc** [comment]
@octavioamu is this a WIP ? If so can we make it onto a draft PR?

**Comment by octavioamu** [comment]
> @octavioamu is this a WIP ? If so can we make it onto a draft PR?

I don't think there is an option to change to draft, just when you create it.

File: app/assets/v2/css/base.css [hunk 3]
```diff
@@ -1054,7 +1056,7 @@ input.is-invalid {
 .rating span.checked label i,
 .rating span.checked ~ span label i {
   color: rgb(18,199,113);
-}
+} */
```
Comment (thelostone-mc) [inline app/assets/v2/css/base.css]: Can we remove out the commented out ?

File: app/assets/v2/css/rating.css [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  -webkit-box-orient: vertical;
+  text-overflow: ellipsis;
+  overflow: hidden;
+  margin-bottom: 6px;
+}
+
+.card-rating__avatar {
+  flex-shrink: 0;
+  /* width: 56px;
+  height: 56px; */
```
Comment (thelostone-mc) [inline app/assets/v2/css/rating.css]: remove?

**Comment by thelostone-mc** [comment]
- Could we make the text area border the same as the other input boxes? [SEE HERE](https://user-images.githubusercontent.com/5358146/55351043-00d46000-54db-11e9-99c1-e17cd9ad5e2d.png)

- Upon giving the feedback while submitting the work -> how to propagate this to the funder ?
I don't see the rating turn up on the profile. [SEE HERE](https://user-images.githubusercontent.com/5358146/55351259-7809f400-54db-11e9-956d-1a2730c44d4b.png)
Also, would we be calculating the average rating and storing it against the User model ?

- The rating doesn't seem to reflect even when the funder rates the contributor. [SEE HERE](https://user-images.githubusercontent.com/5358146/55351445-041c1b80-54dc-11e9-9760-372564c464fa.png)

File: app/assets/v2/js/pages/process_bounty.js [hunk 5]
```diff
@@ -1,6 +1,17 @@
 /* eslint-disable no-console */
 window.onload = function() {
 
+  const changeRateUser = () => {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/process_bounty.js]: could we rename this to rateContributor / addRating to make it more meaningful  ?

File: app/dashboard/views.py [hunk 6]
```diff
@@ -1620,6 +1658,24 @@ def profile(request, handle):
...
+        bounty_owner_github_username=profile.handle,
+        idx_status='done')
+
+    unrated_contributed_bounties = Bounty.objects.current().filter(interested__profile=profile).filter(interested__status='okay') \
+        .filter(interested__pending=False).filter(idx_status='done')
+
+    context['unrated_funded_bounties'] = []
+    context['unrated_contributed_bounties'] = []
+    for bounty in unrated_funded_bounties:
+        if not FeedbackEntry.objects.filter(bounty=bounty, feedbackType='approver'):
```
Comment (danlipert) [inline app/dashboard/views.py]: I think this process of getting all the bounties, then checking each bounty to see if feedback exists is going to perform slowly. We should find a way to just get them in one query for each context entry. Maybe something like: `context['unrated_funded_bounties'] = Bounty.objects.prefetch_related(...,).filter(...,feedbacks__isnull=)` using the `related_name` property of `FeedbackEntry` here: https://github.com/gitcoinco/web/blob/0d76622b37c87da737d2a01aaca11c3f3edfdb5c/app/dashboard/models.py#L3010

File: app/dashboard/templates/bounty/details.html [hunk 7]
```diff
@@ -441,7 +441,7 @@ <h5 class="bounty-heading">{% trans "Funder" %}</h5>
   <script src="{% static "v2/js/tokens.js" %}"></script>
   <script src="{% static "v2/js/amounts.js" %}"></script>
   <script src="{% static "v2/js/popper.min.js" %}"></script>
-  <script src="{% static "v2/js/bootstrap.min.js" %}"></script>
+  <script src="{% static "v2/js/bootstrap.min.js" %}" crossorigin="anonymous"></script>
```
[the remark this answers is not in the document]
Comment (octavioamu) [reply app/dashboard/templates/bounty/details.html]: Is needed for the prefetch tag in the header

**Pull request merged**